src/Features/Core/Portable/PullMemberUp/AbstractMembersPullerService.cs

```diff
@@ -300,10 +310,58 @@ private static ISymbol GetSymbolsToPullUp(MemberAnalysisResult analysisResult)
                 newDestination = destinationEditor.Generator.WithModifiers(newDestination, modifiers);
             }
 
-            destinationEditor.ReplaceNode(destinationSyntaxNode, (syntaxNode, generator) => newDestination);
+            // add imports by moving all source imports to destination container, then taking out unneccessary
+            // imports that we just added.
+            var syntaxFactsService = destinationEditor.OriginalDocument.Project.LanguageServices.GetRequiredService<ISyntaxFactsService>();
+            var destinationTrivia = GetLeadingTriviaBeforeFirstMember(destinationEditor.OriginalRoot, syntaxFactsService);
+            var addImportsService = destinationEditor.OriginalDocument.Project.LanguageServices.GetRequiredService<IAddImportsService>();
+            var semanticModel = await destinationEditor.GetChangedDocument().GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            destinationEditor.ReplaceNode(destinationEditor.OriginalRoot, (node, generator) => addImportsService.AddImports(
+                semanticModel.Compilation,
+                node,
+                node.GetCurrentNode(destinationSyntaxNode),
+                sourceImports,
+                generator,
+                options.PlaceSystemNamespaceFirst,
+                destinationEditor.OriginalDocument.CanAddImportsInHiddenRegions(),
+                cancellationToken));
```

Comment:
so: here are tests i'm hoping we have.  If so, great, if not, please add:

1. tests where the destination has the usings in teh containing namespace, as opposed to the compilation unit.
2. tests where the original code used aliases, and we want the `using alias` to migrate to the new file and not be removed.
3. tests where the using was used as part of an extension method call in teh original code.
4. tests that validate the 'PLaceSystemNamespaceFirst' option.